wagtail/admin/views/generic/models.py

```diff
@@ -241,48 +242,49 @@ def search_queryset(self, queryset):
         }
         return queryset.filter(**filters)
 
-    def _get_title_column(self, column_class=TitleColumn):
-        def title_accessor(obj):
-            draftstate_enabled = self.model and issubclass(self.model, DraftStateMixin)
-
-            if draftstate_enabled and obj.latest_revision:
-                return obj.latest_revision.object_str
-            return str(obj)
-
-        return column_class(
-            "name",
-            label=gettext_lazy("Name"),
-            accessor=title_accessor,
-            get_url=self.get_edit_url,
+    def _get_title_column(self, field_name, column_class=TitleColumn, **kwargs):
+        if not self.model:
+            return column_class(
+                "name",
+                label=gettext_lazy("Name"),
+                accessor=str,
+                get_url=self.get_edit_url,
+            )
+        return self._get_custom_column(
+            field_name, column_class, get_url=self.get_edit_url, **kwargs
         )
 
-    def _get_updated_at_column(self, column_class=DateColumn):
-        return column_class("_updated_at", label=_("Updated"), sort_key="_updated_at")
+    def _get_custom_column(self, field_name, column_class=Column, **kwargs):
+        label, attr = label_for_field(field_name, self.model, return_attr=True)
+        sort_key = getattr(attr, "sort_key", None)
```

Comment:
I'm following the `Column` class from the tables framework to name this `sort_key`, but I realise that `ModelAdmin` currently uses `admin_order_field` instead.

Meanwhile, the `label_for_field` from Django's admin (that we use here) uses `short_description` for the label, which is also used by `ModelAdmin`, as opposed to just label in the `Column` class.

@gasman, Do you think we should rename this to `admin_order_field`, or keep it as it is?

Reply:
@laymonage Is this a pattern that we want to encourage? It seems to me that if you want to customise these details you'd be better off passing an explicit `Column` object in `list_display`, rather than decorating the model method to make the autogenerated one behave as desired.

Having said that, I can see the value of picking those attributes up if someone has already set them for the benefit of ModelAdmin or Django admin - so on that basis, yes, let's rename it to `admin_order_field`. (I think that's a better name to use within model code too - `label` and `sort_key` are fine when they're used inside a column definition, but outside of that context it makes sense to give it a name that explicitly mentions 'admin', so make it clear what it's being used for.)